src/components/connection_handler/test/connection_handler_impl_test.cc

```diff
@@ -412,15 +416,22 @@ TEST_F(ConnectionHandlerTest, GetAppIdOnSessionKey) {
   AddTestSession();
 
   uint32_t app_id = 0;
+  DeviceHandle test_handle;
   const uint32_t test_id = SessionHash(uid_, out_context_.new_session_id_);
 
-  connection_handler::DeviceHandle* device_id = NULL;
   EXPECT_EQ(0,
             connection_handler_->GetDataOnSessionKey(
-                connection_key_, &app_id, NULL, device_id));
+                connection_key_, &app_id, NULL, &test_handle));
   EXPECT_EQ(test_id, app_id);
 }
 
+TEST_F(ConnectionHandlerTest, GetAppIdOnSessionKey_InvalidConnectionId) {
+  uint8_t invalid_id = 0;
+  EXPECT_EQ(
+      -1,
+      connection_handler_->GetDataOnSessionKey(invalid_id, NULL, NULL, NULL));
```

Comment:
@mkorniichuk Since we use the C++ 11 standard, you could use the `nullptr` instead of `NULL` macros.